src/big_core_input.cpp

```diff
@@ -109,29 +109,32 @@ template<typename T>
 const T* BigCoreInput::operator[](size_t index) const
 {
     // RV: tady jsem mel chybu, jako uzivatel chci ukazatel na outermost entity s indexem index (napriklad ukazatel na obrazek cislo 3), abych si z neho mohl dale brat hodnoty co potrebuju
+	// LB: tenhle return ale vrati i vsechny nasledujici obrazky
     return static_cast<const T*>(this->data + outermostEntitiesOffsets[index]);
 }
 
 template<typename T>
 const T& BigCoreInput::at(uint64_t imageNum, uint64_t row, uint64_t column, uint64_t plane, uint64_t tileNum) const
 {
     // RV: musi byt ostre znamenko vetsi. Pokud mam napr. 3 obrazky, indexy jsou 0,1,2.
-	if (imageNum > this->numberOfImages)
+	// LB: musi byt >=, pokud budu v tvem pripade chtit obrazek s indexem 3, tak jsem out of bounds (pokud indexujem od nuly, na coz spoleham)
+	if (imageNum >= this->numberOfImages)
 		throw "Image number out of bound!";
 
-	if (row > this->imageHeight)
+	if (row >= this->imageHeight)
 		throw "Row out of bound!";
 
-	if (column > this->imageWidth)
+	if (column >= this->imageWidth)
 		throw "Column out of bound!";
 
-	if (plane > this->numberOfPlanes)
+	if (plane >= this->numberOfPlanes)
 		throw "Number of planes out of bound!";
 
-	if (tileNum > this->numberOfTiles)
+	if (tileNum >= this->numberOfTiles)
 		throw "Tile number out of bound!";
 
-    // RV: TODO: overit, zda jsou data v pameti a pokud ne, tak je nacist
+    // RV: TODO: overit, zda jsou data v pameti a pokud ne, tak je nacist 
+	// LB: odkud?
```

Comment:
Ze souboru. V soucasne verzi to znamena pouze volat loadToMemory( ). Overime to tedy pouze tim, ze dataLength > 0. V budoucnu az to budeme rozsirovat na vetsi soubory nez mame pamet, bude potreba overit, zda je obrazek v bufferu a pokud ne, nacist pouze tento jeden obrazek do bufferu a z nej data vratit.